core/gather/fetcher.js

```diff
@@ -25,34 +25,58 @@ class Fetcher {
    * Fetches any resource using the network directly.
    *
    * @param {string} url
-   * @param {{timeout: number}=} options timeout is in ms
+   * @param {{timeout?: number, responseHeaders?: string[]}=} options
+   *  - timeout is in ms
+   *  - responseHeaders is used to filter the resulting headers object.
+   *  Sometime in the future, when Node has full Fetch support, this can be removed
+   *  and instead we can just return a Headers instance. In the meantime, ask users
+   *  to early-declare the headers they are interested in so we can skip some processing
+   *  in the `_fetchWithFetchApi` case.
    * @return {Promise<FetchResponse>}
    */
-  async fetchResource(url, options = {timeout: 2_000}) {
+  async fetchResource(url, options = {}) {
+    const resolvedOptions = {
+      timeout: options.timeout ?? 2_000,
+      responseHeaders: options.responseHeaders ?? [],
+    };
+
     // In Lightrider, `Network.loadNetworkResource` is not implemented, but fetch
     // is configured to work for any resource.
     if (global.isLightrider) {
-      return this._wrapWithTimeout(this._fetchWithFetchApi(url), options.timeout);
+      return this._wrapWithTimeout(
+        this._fetchWithFetchApi(url, resolvedOptions), resolvedOptions.timeout);
     }
 
-    return this._fetchResourceOverProtocol(url, options);
+    return this._fetchResourceOverProtocol(url, resolvedOptions);
   }
 
   /**
    * @param {string} url
+   * @param {{responseHeaders: string[]}} options
    * @return {Promise<FetchResponse>}
    */
-  async _fetchWithFetchApi(url) {
+  async _fetchWithFetchApi(url, options) {
     const response = await fetch(url);
 
     let content = null;
     try {
       content = await response.text();
     } catch {}
 
+    /** @type {LH.Crdp.Network.Headers|null} */
+    let headers = null;
+    if (response.headers) {
+      headers = {};
+      for (const header of options.responseHeaders) {
+        const value = response.headers.get(header);
+        if (value !== null) headers[header] = value;
+      }
+    }
```

Comment:
fetch spec's got your back :)
```suggestion
    const headers = Object.fromEntries(response.headers);
```